frontend/src/store.js

```diff
@@ -43,20 +47,101 @@ export default new Vuex.Store({
     ...nativeLanguageState,
     ...teacherState,
     ...screeningState,
+    ...matchingState
     ...reasonState
   },
   mutations: {
     ...studentMutations,
     ...nativeLanguageMutations,
     ...teacherMutations,
     ...screeningMutations,
+    ...matchingMutations
     ...reasonMutations
   },
   actions: {
     ...studentActions,
     ...nativeLanguageActions,
     ...teacherActions,
     ...screeningActions,
+    ...matchingActions,
+
+    // Update student status
+    async updateStudentStatus({ commit, dispatch }, { studentID, previousStatusString, nextStatusString, updatedBy }) {
+      // PATCH student
+      const studentRequestOptions = {
+        method: "PATCH",
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({StudentID: studentID, StatusString: nextStatusString})
+      }
+
+      // POST statusUpdate
+      const statusUpdateRequestOptions = {
+        method: "POST",
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          StudentID: studentID,
+          PreviousStatusString: previousStatusString,
+          NextStatusString: nextStatusString,
+          UpdatedBy: updatedBy
+        })
+      }
+
+      const patchStudent = fetch("/api/students/" + studentID, studentRequestOptions)
+      const postStatusUpdate = fetch("/api/statusUpdates", statusUpdateRequestOptions)
+
+      Promise.all([
+        patchStudent,
+        postStatusUpdate
+      ])
+      .then(([responsePatch, responsePost]) => {
+          // If PATCH and/or POST fail, return
+          if(responsePatch.status !== 200) {
+            console.log('responsePatch', responsePatch)
+            return;
+          }
+          if(responsePost.status !== 200) {
+            console.log('responsePost', responsePost)
+            return;
+          }
+          dispatch('getAllStudents')
+      }).catch((err) => {
+        console.error(err);
+      });
+    },
+
+    // Update student English proficiency
+    async updateStudentEnglishProficiency({ commit, dispatch }, { studentID, englishProficiency }) {
+
+      if (!englishProficiency) return;
+
+      // PATCH student
+      const studentRequestOptions = {
+        method: "PATCH",
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({StudentID: studentID, EnglishProficiency: englishProficiency})
+      }
+
+      fetch("/api/students/" + studentID, studentRequestOptions)
+        .then(response => {
+            // If PATCH fails, return
+            if(response.status !== 200) {
+              console.log(response);
+              return;
+            }
+            dispatch('getAllStudents')
+          }
+        )
+    },
+
```

Comment:
The `updateStudentStatus()` and `updateStudentEnglishProficiency()` actions are already present in `frontend/src/store/students.js` under `studentActions`. (Do verify this for yourself.)

```suggestion
```